File: python/gcd.py
def gcd(x, y):
    if x > y:
        small = y
    else:
        small = x
    for i in range(1, small+1):
        if((x % i == 0) and (y % i == 0)):
            hcf = i 
    return hcf

num1 = int(input("enter num1: "))
num2 = int(input("enter num2: "))

print("The H.C.F. is", gcd(num1, num2))

File: java/GCD.java
import java.util.Scanner;
public class GCD{
    public static void main(String[] args){
        int num1, num2;
        Scanner myobj = new Scanner(System.in);
        System.out.print("Enter first number: ");
        num1 = myobj.nextInt();

        System.out.print("Enter second number: ");
        num2 = myobj.nextInt();
        myobj.close()
        while(num1 != num2){
            if(num1>num2){
                num1 = num1-num2;
            }
            else{
                num2 = num2 - num1
            }

        }
        System.out.println("GCD of given numbers is: %d", num2);
    }
}